logspout/logspout.go

```diff
@@ -102,9 +102,11 @@ func getLogName(name string) (string, string) {
 	if match != nil {
 		return match[1], match[3]
 	}
-	match = getMatch(`^k8s_([a-z0-9-]+)-([a-z]+)\.`, name)
```

Comment:
I believe if you use `^k8s_([a-z0-9-]+)-[a-z]+\.[\da-f]+_[a-z0-9-]+-([a-z]+-[\da-z]*)_` you can forgo all the additional logic you are adding in subsequent lines.  It would capture two groups (like it does already), and they will both have the right stuff in them.  You'll end up with log messages that look like the following:

```
2015-08-26T23:25:45UTC dashing-picard[web-gw7pp]: 2015/08/26 23:25:45 10.244.48.0:43701 GET /
2015-08-26T23:25:45UTC dashing-picard[web-gw7pp]: 2015/08/26 23:25:45 10.244.48.0:43703 GET /favicon.ico
```

I suppose the dashes in the "pid" could then be converted to periods if we feel that's important, but that's a one liner.